src/main/java/org/apache/sysml/hops/FunctionOp.java

```diff
@@ -33,14 +34,14 @@
 /**
  * This FunctionOp represents the call to a DML-bodied or external function.
  * 
- * Note: Currently, we support expressions in function arguments but no function calls
- * in expressions.
+ * Note: Currently, we support expressions in function arguments along with function calls
+ * in expressions with single outputs, leaving multiple outputs handling as it is.
  */
 public class FunctionOp extends Hop
 {
 	
 	public static String OPSTRING = "extfunct";
-	
+	public static boolean isSingleFunctionInvoked = false;
```

Comment:
Please make this a member variable (not static), and pass it as parameter into the constructor. For now, always call it with false. Also maybe shorten this name a bit.